[Bug] GemmTest error

cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug -DGLOW_WITH_OPENCL=0 ../glow

"ninja all" seems OK:
[1/167] Building CXX object lib/Backends/CPU/CMakeFiles/CPURuntime.dir/libjit/libjit.cpp.o
[2/167] Building CXX object lib/Backends/CPU/CMakeFiles/CPURuntime.dir/libjit/libjit_conv.cpp.o
[3/167] Building CXX object lib/Backends/CPU/CMakeFiles/CPURuntime.dir/libjit/libjit_matmul.cpp.o
[4/167] Building CXX object lib/Backends/CPU/CMakeFiles/CPURuntimeNative.dir/libjit/libjit.cpp.o
... skip ...
[162/167] Building CXX object tests/unittests/CMakeFiles/GemmTest.dir/GemmTest.cpp.o
[163/167] Building CXX object tests/unittests/CMakeFiles/onnxImporterTest.dir/onnxImporterTest.cpp.o
[164/167] Linking CXX executable tests/caffe2ImporterTest
[165/167] Linking CXX executable tests/BackendCorrectnessTest
[166/167] Linking CXX executable tests/GemmTest
In file included from /home/walker/glow/glow/tests/unittests/onnxImporterTest.cpp:21:0:
/home/walker/glow/glow/tests/googletest/googletest/include/gtest/gtest.h: In instantiation of ‘testing::AssertionResult testing::internal::CmpHelperEQ(const char*, const char*, const T1&, const T2&) [with T1 = long unsigned int; T2 = int]’:
/home/walker/glow/glow/tests/googletest/googletest/include/gtest/gtest.h:1421:23:   required from ‘static testing::AssertionResult testing::internal::EqHelper<lhs_is_null_literal>::Compare(const char*, const char*, const T1&, const T2&) [with T1 = long unsigned int; T2 = int; bool lhs_is_null_literal = false]’
/home/walker/glow/glow/tests/unittests/onnxImporterTest.cpp:53:3:   required from here
/home/walker/glow/glow/tests/googletest/googletest/include/gtest/gtest.h:1392:11: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (lhs == rhs) {
           ^
[167/167] Linking CXX executable tests/onnxImporterTest


ninja test:
[1/1] Running tests...
Test project /home/walker/glow/build_Debug
      Start  1: partitionTest
 1/20 Test  #1: partitionTest ....................   Passed    0.05 sec
      Start  2: tensorsTest
 2/20 Test  #2: tensorsTest ......................   Passed    0.43 sec
      Start  3: gradCheckTest
 3/20 Test  #3: gradCheckTest ....................   Passed    4.04 sec
      Start  4: IROptTest
 4/20 Test  #4: IROptTest ........................   Passed    0.08 sec
      Start  5: basicIRTest
 5/20 Test  #5: basicIRTest ......................   Passed    0.07 sec
      Start  6: backendTest
 6/20 Test  #6: backendTest ......................   Passed    1.32 sec
      Start  7: MLTest
 7/20 Test  #7: MLTest ...........................   Passed   22.99 sec
      Start  8: operatorTest
 8/20 Test  #8: operatorTest .....................   Passed    4.51 sec
      Start  9: graphTest
 9/20 Test  #9: graphTest ........................   Passed    0.83 sec
      Start 10: graphGradTest
10/20 Test #10: graphGradTest ....................   Passed    0.03 sec
      Start 11: graphOptzTest
11/20 Test #11: graphOptzTest ....................   Passed    0.18 sec
      Start 12: graphSchedulerTest
12/20 Test #12: graphSchedulerTest ...............   Passed    0.06 sec
      Start 13: quantizationTest
13/20 Test #13: quantizationTest .................   Passed    1.08 sec
      Start 14: UtilsTest
14/20 Test #14: UtilsTest ........................   Passed    0.04 sec
      Start 15: BackendCorrectnessTest
15/20 Test #15: BackendCorrectnessTest ...........   Passed   14.09 sec
      Start 16: GemmTest
16/20 Test #16: GemmTest .........................***Failed    0.03 sec
      Start 17: LLVMIRGenTest
17/20 Test #17: LLVMIRGenTest ....................   Passed    0.00 sec
      Start 18: memoryAllocatorTest
18/20 Test #18: memoryAllocatorTest ..............   Passed    0.25 sec
      Start 19: caffe2ImporterTest
19/20 Test #19: caffe2ImporterTest ...............   Passed    0.09 sec
      Start 20: onnxImporterTest
20/20 Test #20: onnxImporterTest .................   Passed    0.01 sec

95% tests passed, 1 tests failed out of 20

Total Test time (real) =  50.25 sec

The following tests FAILED:
	 16 - GemmTest (Failed)
FAILED: cd /home/walker/glow/build_Debug && /usr/local/bin/ctest --force-new-ctest-process
ninja: build stopped: subcommand failed.

Why the GemmTest failed?  Can you help me?

--- a/lib/Backends/CPU/libjit/libjit_conv.cpp
+++ b/lib/Backends/CPU/libjit/libjit_conv.cpp
@@ -63,7 +63,7 @@ void libjit_convDKKC8_convolve_channel(
   float8 sum[numDepthRegs][ywidth];
   for (unsigned wu = 0; wu < ywidth; wu++) {
     for (unsigned du = 0; du < numDepthRegs; du++) {
-      sum[du][wu] = 0;
+      sum[du][wu] = BroadcastFloat8(0.0);
     }
   }
 
@@ -77,7 +77,7 @@ void libjit_convDKKC8_convolve_channel(
       // Load a single pixel from the input image and broadcast it.
       auto inIdx = libjit_getXYZW(inWdims, sampleN, inX, inY + wu * stride,
                                   fd + group * numChannels);
-      in8[wu] = inW[inIdx];
+      in8[wu] = BroadcastFloat8(inW[inIdx]);
     }
 
     // For each y pixel:

--- a/lib/Backends/CPU/libjit/libjit_defs.h
+++ b/lib/Backends/CPU/libjit/libjit_defs.h
@@ -24,8 +24,13 @@
 typedef SSIZE_T ssize_t;
 #endif
 
-typedef float float4 __attribute__((ext_vector_type(4)));
-typedef float float8 __attribute__((ext_vector_type(8)));
+#if defined(__clang__)
+using float4 = float __attribute__((ext_vector_type(4)));
+using float8 = float __attribute__((ext_vector_type(8)));
+#elif defined(__GNUC__) || defined(__GNUG__)
+using float4 = float __attribute__((vector_size(16)));
+using float8 = float __attribute__((vector_size(32)));
+#endif
 
 /// Loads a simd float8 value from \p ptr.
 #define LoadFloat8(PTR) *((const float8 *)(PTR))
@@ -37,7 +42,11 @@ typedef float float8 __attribute__((ext_vector_type(8)));
 #define AddFloat8(PTR, VAL) *((float8 *)(PTR)) += (VAL);
 
 /// Broadcast the input value to a float8.
+#if defined(__clang__)
 #define BroadcastFloat8(VAL) ((float8)(VAL))
+#elif defined(__GNUC__) || defined(__GNUG__)
+#define BroadcastFloat8(VAL) ((VAL) - (float8){})
+#endif
 
 #define MIN(a, b) (((a) < (b)) ? (a) : (b))
 #define MAX(a, b) (((a) > (b)) ? (a) : (b))

